### PR TITLE
Select fields for KeyTable

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

**Comment by jigold** [comment]
@tpoterba It would be great if you could look at this -- let me know if you don't have time and I'll assign to someone else. @cseed and I discussed KeyTable select last week and decided we should keep the regular select, but also have a separate `select_fields` method.

File: python/hail/keytable.py [hunk 1]
```diff
@@ -636,6 +636,76 @@ def select(self, selected_columns):
 
     @handle_py4j
     @record_method
+    @typecheck_method(selected_fields=oneof(strlike, listof(strlike)),
```
Comment (cseed) [inline python/hail/keytable.py]: You should delete export_samples, export_variants and export_genotypes and use this + KeyTable.export.

**Comment by tpoterba** [comment]
Why are we keeping both if they're almost exactly the same thing? To prep for the python interface?

**Comment by cseed** [comment]
Am I misremembering what I said?  (Probably.)  I thought this was going to replace select, and select was going to become "create a new schema from scratch", whereas annotate is add to existing schema.  Finally, we should eventually add transmute, which adds new fields but drops anything mentioned on the RHS, but leaves unmentioned fields unchanged.

**Comment by jigold** [comment]
Yes, all of that (export changes, transmute, and updated select) was part of my plan. I was trying to keep the PRs short. This one is already 250 lines. If the issue is the method name `select_fields`, then I can change that now.

**Comment by jigold** [comment]
Closing this for now. Want to rethink this interface.

**Comment by jigold** [comment]
Closing in favor of #2279

**Pull request closed without merging**